### PR TITLE
add updated script from MSOP10

### PR DESCRIPTION
**Pull request by lukh**
Providing HVSSOP yaml settings for https://github.com/KiCad/kicad-symbols/pull/2717

### CODE REVIEW

**Comment by cpresser** [comment]
Related: https://github.com/KiCad/kicad-footprints/pull/2460

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  overall_size_x:
+    nominal: 4.9
+    tolerance: 0.152
+  lead_width:
+    minimum: 0.17
+    maximum: 0.27
+  lead_len:
+    nominal: 0.53
+    tolerance: 0.152
```
Comment (cpresser) [inline scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml:25]: Where did you get that value from?

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    tolerance: 0.100
+  body_size_y:
+    nominal: 3
+    tolerance: 0.100
+  overall_height:
+    maximum: 1.1
+
+  overall_size_x:
+    nominal: 4.9
+    tolerance: 0.152
```
Comment (cpresser) [inline scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml:19]: I see 5.05 .. 4.75 No clue where the 0.152 value comes from.
You can just enter them as
```yaml
minimum 4.75
maximum 5.05
```

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    nominal: 0.53
+    tolerance: 0.152
+
+  pitch: 0.5
+  num_pins_x: 0
+  num_pins_y: 5
+
+  EP_size_x:
+    nominal: 1.57
+    tolerance: 0
```
Comment (cpresser) [inline scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml:33]: please remove the tolerance:0 line
Also remove the un-commented lines while you are at it.

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  thermal_vias:
+    count: [2, 2]
+    drill: 0.2
+    # min_annular_ring: 0.15
+    paste_via_clearance: 0.1
+    EP_num_paste_pads: [2, 2]
+    # paste_between_vias: 1
+    # paste_rings_outside: 1
+    EP_paste_coverage: 0.7
+    grid: [1.1, 1.3]
```
Comment (cpresser) [inline scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml:53]: Can you elaborate why you choose any of those value?
Why 2x2 paste pads? Why the 0.7 coverage, why selecting that grid?

**Review by cpresser — CHANGES_REQUESTED** [review]
The exposed pad looks weird. I can't find any of the values used there in the datasheet.

AFAIK there is no way to get the same via layout as in the datasheet, but I don't think that is required. 2x2 layout for vias seems okay.

**Comment by lukh** [comment]
Hi, thank for the help;
a general remark based on yours,
The work here is mainly based on the MSOP-10 package, 
regarding this " Very Thin Shrink Small Outline Package, also referred as MSOP = Micro Small Outline Package " (https://www.ti.com/support-packaging/packaging-resources/packaging-terminology.html)

And the fact that the component designed (https://github.com/KiCad/kicad-symbols/pull/2717), in its first revision, was using the MSOP-10 package, with sucess.

I work on the specific points now !
Bests,

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml (hunk 1 above)
Comment (lukh) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml:25]: This one was from the MSOP-10 (MSOP-10-1EP_3x3mm_P0.5mm_EP1.68x1.88mm_ThermalVias)

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml (hunk 2 above)
Comment (lukh) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml:19]: Done

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml (hunk 4 above)
Comment (lukh) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml]: Well, same as the other... The MSOP-10 did work well on the bq24090, so I used this one.

- I don't know if it is possible to have a fifth thermal vias, at the centre

Comment (cpresser) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml]: I looked at the generator and could not find a feature for the 5th center thermal via.
But the 4 generated vias, when using the default settings are fine. They are according to IPC recommendations.

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml (hunk 1 above)
Comment (cpresser) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml:25]: I found `0.4 .. 0.7` in the datasheet.

**Comment by cpresser** [comment]
Hi,
sorry for all the fuzz, I guess from your point of view that is pretty ridiculous.

> The work here is mainly based on the MSOP-10 package,
> regarding this " Very Thin Shrink Small Outline Package, also referred as MSOP = Micro Small Outline Package " (https://www.ti.com/support-packaging/packaging-resources/packaging-terminology.html)
> 
> And the fact that the component designed ([KiCad/kicad-symbols#2717](https://github.com/KiCad/kicad-symbols/pull/2717)), in its first revision, was using the MSOP-10 package, with sucess.
>

There are two reasons why I am picky here:
1. Consistency. People looking at this in the future should not ask themselves why some dimensions differ from the datasheet. Also, why make a new footprint if its the same as the existing one?
2. We want footprints to be as good as possible - to maximize yield in automated production lines. 


Anyway, I like you changes. There is just one thing I still don't understand. Why the line 
```yaml
grid: [1.1, 1.3]
```
What does it do?

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml (hunk 4 above)
Comment (lukh) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml]: ok great !

**Comment by lukh** [comment]
Hi,
No problems at all ! It is a good thing :+1: 

Ok the grid apply a coefficient to dimensions of the back pad of EP.
I remove it.

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml (hunk 4 above)
Comment (lukh) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml]: You mean, should I remove all param except count ?

Comment (lukh) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml]: Removed all non necessary params

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml (hunk 1 above)
Comment (lukh) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/hvssop.yaml]: Updated

**Comment by cpresser** [comment]
Looks good. Thanks!

**Pull request merged**